roles/multus-crd/tasks/main.yml

```diff
@@ -0,0 +1,71 @@
+---
+
+- name: Create folder for Multus resource defs
+  file:
+    path: "{{ ansible_env.HOME }}/multus-resources"
+    state: directory
+
+- name: Template multus resources
+  template:
+    src: "{{ item.src }}"
+    dest: "{{ ansible_env.HOME }}/multus-resources/{{ item.dest }}"
+  with_items:
+    - src: crd.yml.j2
+      dest: "multus-crd.yml"
+    - src: flannel.yml.j2
+      dest: "flannel.yml"
+    - src: macvlan.yml.j2
+      dest: "macvlan.yml"
+
+- name: Check to see if CRD is present
+  shell: >
+    kubectl get crd
+  register: check_crd
+
+- name: Create base CRD
+  shell: >
+    kubectl create -f {{ ansible_env.HOME }}/multus-resources/multus-crd.yml
+  when: "'networks.kubernetes.com' not in check_crd.stdout"
+
+- name: Check to see which network CRD definitions are present
+  shell: >
+    kubectl get networks
+  register: check_network_crds
+
+- name: Create flannel network CRD
+  shell: >
+    kubectl create -f {{ ansible_env.HOME }}/multus-resources/flannel.yml
+  when: "'flannel-conf' not in check_network_crds.stdout"
+
+- name: Create macvlan network CRD
+  shell: >
+    kubectl create -f {{ ansible_env.HOME }}/multus-resources/macvlan.yml
+  when: "'macvlan-conf' not in check_network_crds.stdout"
+
+- name: Check to see which CRDs are present, for validation
```

Comment:
Good idea, here's where I list the "networks" from the CRDs, and then check against the names just to be complete there.